docs/whitepaper.tex

```diff
@@ -220,9 +247,118 @@ \subsection{DEX pool}
 \end{itemize}
 
+
+\subsection{Pool Addresses validator}
+
+The pool addresses validator controls when pools can be spent.
+It is therefore responsible for controlling swaps,
+liquidity deposits and withdrawals, pool termination
+and admin fee collection.
+It is also responsible for ensuring that
+pool id tokens never leave its addresses
+because they could then be used to open invalid pools.
+
+To do this it enforces the following:
+Note: ``The output pool" refers to
+the pool in the outputs which has the same pool id
+as the input pool invoking the validator.
+
+\begin{itemize}
+	\item The input and output pools are valid
+	\item The redeemer is valid
+	\item When the redeemer action is Swap:
+		\begin{itemize}
+			\item The exchange invariant doesn't decrease before fees
+			\item The Admin fee is increased by $0.3\%$ of the output
+			\item The pool increases by $0.3\%$ of the output
+			\item No liquidity tokens are minted
+			\item The input and output pools are both live
+		\end{itemize}
+	\item When the redeemer action is Liquidity
+		\begin{itemize}
+			\item The liquidity invariant $L^2/k^2$ is non decreasing before fees
+			and appropriate fees are paid.
+				\todo{how should fees work here?}
+			\item The datum correctly reflects the change in issued liquidity
+			\item The ``liveness" of the pool is not changed
+			\item If the input pool is dead the issued liquidity decreases
+			\item If the input pool is dead the withdraw is balanced
+		\end{itemize}
+	\item When the redeemer action is Kill
+		\begin{itemize}
+			\item The admin signed the transaction
+			\item The output pool is dead
+			\item Nothing else is changed in the pool
+		\end{itemize}
+	\item When The redeemer action is Collect
+		\begin{itemize}
+			\item The admin fees in the output pool are zero
+			\item the non admin fees balance stays the same
+			\item the output value remains correct for the datum
+			\item The admin keys signed the tx
+		\end{itemize}
+\end{itemize}
+
+
+\subsection{Pool id minting policy}
+
+The pool id token is a token used to track pools
+and to ensure that the pools the protocol interacts with were opened legitimately.
+The pool id token's minting policy is responsible for ensuring
+that it is only minted as part of opening a valid vault.
+To do so it should enforce all of the following:
+
+\begin{itemize}
+	\item The config utxo is valid
+	\item Exactly one token is minted
+	\item That token is sent to a pool
+	\item The redeemer provides a seed utxo
+	\item The token name of the minted token is the hash of the seed utxo
+	\item The seed utxo is spent in the transaction
+	\item The output pool datum is valid
+	\item The datum correctly reflects the number of liquidity tokens being minted
+	\item at least one liquidity token is being minted
+	\item the initial invariant is not zero ie. some liquidity is provided in each token
+	\item The issued liquidity is exactly the square root of the initial value
```

Comment:
What is "the initial value"? Feel free to repeat parts from earlier in the spec here.

Reply:
What does "exactly" mean, here? It's very unlikely that the square root of `x_0 \cdot y_0` is an integer.